cli/actions/transaction.go

```diff
@@ -0,0 +1,135 @@
+package actions
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	yamllib "github.com/goccy/go-yaml"
+	"github.com/kubeshop/tracetest/cli/file"
+	"github.com/kubeshop/tracetest/cli/openapi"
+	"github.com/kubeshop/tracetest/cli/utils"
+	"github.com/kubeshop/tracetest/server/model/yaml"
+)
+
+type transactionActions struct {
+	resourceArgs
+
+	openapiClient *openapi.APIClient
+}
+
+var _ ResourceActions = &transactionActions{}
+
+func NewTransactionsActions(openapiClient *openapi.APIClient, options ...ResourceArgsOption) transactionActions {
+	args := NewResourceArgs(options...)
+
+	return transactionActions{
+		resourceArgs:  args,
+		openapiClient: openapiClient,
+	}
+}
+
+// Apply implements ResourceActions
+func (actions transactionActions) Apply(ctx context.Context, file file.File) (*file.File, error) {
+	newFile, err := actions.convertTestPathToId(ctx, file)
+	if err != nil {
+		return nil, fmt.Errorf("could not process tests from transaction: %w", err)
+	}
+
+	if newFile.HasID() {
+		rawTransaction, err := actions.formatter.ToStruct(&newFile)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse file into transaction: %w", err)
+		}
+
+		transaction := rawTransaction.(openapi.TransactionResource)
+		return actions.resourceClient.Update(ctx, newFile, *transaction.Spec.Id)
```

Comment:
won't this fail if the file has an ID but it's not yet created? I think this is what happens with environments

Reply:
Indeed we receive a 404 when we don't have an ID.